server/src/main/java/org/elasticsearch/action/search/SearchShardsResponse.java

```diff
@@ -80,4 +94,22 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(groups, nodes, aliasFilters);
     }
+
+    public static SearchShardsResponse fromLegacyResponse(ClusterSearchShardsResponse oldResp) {
```

Comment:
does this need to be public?

Reply:
it looks to me like this could be package private.

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/95894/commits/8f5b715aaeac886f0d4f1551141ea5b0d067921b